### PR TITLE
fix(Scripts/Ulduar): Yogg-Saron's portals

### PR DESCRIPTION
**Pull request by Si1ker**
This change will build the structure and allocate the portals correctly

## CHANGES PROPOSED:
-  Add structure, locations and allocation method

## ISSUES ADDRESSED:
- N/A

## TESTS PERFORMED:
- Tested in-game

## HOW TO TEST THE CHANGES:

Go Ulduar, kill keepers, go Yoggsaron and take it to Phase 2, see how portals spawn around homogeneously

## KNOWN ISSUES AND TODO LIST:

- N/A

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp [hunk 1]
```diff
@@ -296,6 +298,26 @@ enum Misc
     DATA_GET_DRIVE_ME_CRAZY             = 4,
 };
 
+struct LocationsXY
+{
+    float x, y, z, o;
+    uint32 id;
+};
+
+static LocationsXY YoggPortalLoc[] =
```
Comment (FrancescoBorzi) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp:307]: `YoggPortalLoc` should start with lowercase

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp [hunk 2]
```diff
@@ -296,6 +298,26 @@ enum Misc
     DATA_GET_DRIVE_ME_CRAZY             = 4,
 };
 
+struct LocationsXY
+{
+    float x, y, z, o;
+    uint32 id;
```
Comment (FrancescoBorzi) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp:304]: why `uint32`? also where is this `id` used?

Comment (Si1ker) [reply src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp]: oops 🐱

**Comment by Si1ker** [comment]
[ 98%] Building CXX object src/server/scripts/CMakeFiles/scripts.dir/Pet/pet_shaman.cpp.o
/home/runner/work/azerothcore-wotlk/azerothcore-wotlk/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp:549:24: fatal error: using the result of an assignment as a condition without parentheses [-Wparentheses]
                if (cr = me->SummonCreature(NPC_DESCEND_INTO_MADNESS, yoggPortalLoc[i].x, yoggPortalLoc[i].y, yoggPortalLoc[i].z, 0, TEMPSUMMON_TIMED_DESPAWN, 25000))
                    ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/azerothcore-wotlk/azerothcore-wotlk/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp:549:24: note: `place parentheses around the assignment to silence this warning`

well I was hoping to avoid it but let's make build on ubuntu passing, adding them back.

**Comment by iyasenai** [comment]
It works, watch the video :).
https://www.youtube.com/watch?v=UnxThoj4gI4    (10)
https://www.youtube.com/watch?v=m_h0hM8ZBco (25)

**Comment by Si1ker** [comment]
Thanks @iyasenai really appreciated! ⭐
Excellent videos and tests 🚀

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp [hunk 3]
```diff
@@ -296,6 +298,25 @@ enum Misc
     DATA_GET_DRIVE_ME_CRAZY             = 4,
 };
 
+struct LocationsXY
+{
+    float x, y, z, o;
+};
+
+static LocationsXY yoggPortalLoc[] =
```
Comment (FrancescoBorzi) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp:306]: Are we sure there is not already such a structure? If not, we might define a generic one so we can always reuse the same.

Approving the PR already, as the above can be done even later

**Review by FrancescoBorzi — APPROVED** [review]

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp (hunk 3 above)
Comment (FrancescoBorzi) [reply src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp:306]: ah, also the `o` is never used here. @Si1ker please remove it.

Comment (Si1ker) [reply src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp:306]: @FrancescoBorzi change applied.

**Pull request merged**